src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -54,19 +52,32 @@ public static class AlternativeSavingCalculator
 
                 }
             }
-            var approximateSavedAmount = 0M;
+            var timesToRepeatSaving = 0;
+            var approximateSavedAmount = startingBalance;
             while (goal.Amount > approximateSavedAmount)
             {
                 for (var i = enumCount; i > (int)Importance.Necessary; i--)
                 {
-                    averagesOfAmountByImportanceAdjusted[i - 1] = sumsOfAmountByImportanceAdjusted[i - 1] / entryAmounts[i - 1];
-                    averagesOfAmountByImportanceDefault[i - 1] = sumsOfAmountByImportanceDefault[i - 1] / entryAmounts[i - 1];
+                    if (entryAmounts[i - 1] != 0)
+                    {
+                        averagesOfAmountByImportanceAdjusted[i - 1] = sumsOfAmountByImportanceAdjusted[i - 1] / entryAmounts[i - 1];
+                        averagesOfAmountByImportanceDefault[i - 1] = sumsOfAmountByImportanceDefault[i - 1] / entryAmounts[i - 1];
+                    }
+                    else
+                    {
+                        averagesOfAmountByImportanceAdjusted[i - 1] = 0;
+                        averagesOfAmountByImportanceDefault[i - 1] = 0;
+                    }
 
                     approximateSavedAmount += averagesOfAmountByImportanceDefault[i - 1] - averagesOfAmountByImportanceAdjusted[i - 1];
-
-                    monthlySuggestions.Add(new SavingSuggestionByMonth(averagesOfAmountByImportanceAdjusted[i - 1], averagesOfAmountByImportanceDefault[i - 1], (Importance)i));
-                }                              
+                }
+                timesToRepeatSaving++;              
+            }
```

Comment:
For example this looks like it could be it's own separate method

Reply:
the problem is that the front would then need to call two seperate methods, no? where one is just a part of the whole, so I'm not sure about it. Or just seperate the method out and still call it, which, i feel like only adds to unreadability in this case.